Products/ZenHub/services/DiscoverService.py

```diff
@@ -303,3 +303,24 @@ def remote_moveDevice(self, dev, path):
     def remote_getDefaultNetworks(self):
         monitor = self.dmd.Monitors.Performance._getOb(self.instance)
         return [net for net in monitor.discoveryNetworks]
+
+    @translateError
+    def remote_removeNetAppInterfaces(self, net):
+        """
+        Remove IPs for particular network
+        already assigned to NetApp interfaces (nodes)
+
+        @param net - network to discover
+        @return:  a list of IPs without IP assigned for NetApp interfaces
+        @rtype: list
+        """
+
+        full_ip_list = net.fullIpList()
+
+        for d in self.dmd.Devices.getSubDevices():
+            for interface in d.os.interfaces():
+                if '/Storage/NetApp' in interface.getDeviceClassName() and interface.getIpAddressObj():
+                    if net.netmask == interface.getIpAddressObj().netmask and interface.getIp() in full_ip_list:
```

Comment:
you should only call getIpAddressObj once, and the ip can be pulled from the IpAddress object instead of the interface.